### DISCUSSION TITLE
Add device_id as Identifier

### DISCUSSION THREAD

**Discussion by Wonko52** [pqrs-org/Karabiner-Elements#3980] [Ideas]
Hi,

I would find it helpful to use the device_id in the device_if condition.

Background: I use my MacBook at work and at home with a monitor to which a Logitech Unifiying USB receiver is connected. Different keyboard models are connected to it, for which different rules are required.  In the Karabiner EventViewer, the keyboards are recognized via the USB receiver, i.e. product_id and vendor_id are identical. 

As a workaround, I am currently using the location_id, but this always changes when I use a different USB port.

Work (Logitech K450):
```
    {
        "device_id": 4295048935,
        "device_identifiers": {
            "is_keyboard": true,
            "product_id": 50475,
            "vendor_id": 1133
        },
        "location_id": 34668544,
        "manufacturer": "Logitech",
        "product": "USB Receiver",
        "transport": "USB"
    }
```

Home (Logitech Craft):
```
    {
        "device_id": 4295097753,
        "device_identifiers": {
            "is_keyboard": true,
            "product_id": 50475,
            "vendor_id": 1133
        },
        "location_id": 1179648,
        "manufacturer": "Logitech",
        "product": "USB Receiver",
        "transport": "USB"
    }
```

**Comment by tekezo** [comment]
The device_id changes every time the device is connected, so it isn't suitable for this purpose.

For now, the only options seem to be either:
- Using different profiles
- Switching with `variable_if` using variables.
